TPLs.cmake

```diff
@@ -53,7 +53,6 @@ set(PSTREAMS_ROOT ${TPL_DIR}) # prefer ours
 find_package(PStreams REQUIRED)
 
 #### Hypre
-set(HYPRE_ROOT ${TPL_DIR}) # prefer ours
```

Comment:
How do you make it find hypre from TPL now?

Reply:
If the TPL build builds it, it finds ours even without setting `HYPRE_ROOT`, even if it is available on the system.

This now triggers `NO_DEFAULT_PATH` if defined, so that if both the system has it as well as we also have built it, ours is preferred. (For an AMPI-build, even if the system has it we need to build our own. Hence the slightly more complex cmake logic immediately before all `find_package` calls to MPI TPLs.)

Reply:
I think it finds ours because cmake prioritizes `HINTS` before system paths.

Reply:
You don't set `HYPRE_ROOT` and hence `NO_DEFAULT_PATH` isn't set either, so what is used as a hint now? Are we setting an environment variable?

Reply:
We add `TPL_DIR` to `CMAKE_PREFIX_PATH` at https://github.com/quinoacomputing/cmake-modules/blob/master/TPLs.cmake#L11.

Reply:
Thanks!

Reply:
Nice